tests/rule_tests/remove_unused_variable.rs

```diff
@@ -0,0 +1,105 @@
+use darklua_core::rules::{RemoveUnusedVariable, Rule};
+
+test_rule!(
+    remove_unused_variable,
+    RemoveUnusedVariable::default(),
+    remove_unused_local("local foo = true") => "",
+    remove_two_unused_local("local foo, bar = true") => "",
+    remove_unused_local_function("local function foo() end") => "",
+    remove_unused_local_function_single_statement("local function foo() end") => "",
+    remove_unused_local_function_recursive("local function foo() foo() end") => "",
+    remove_function_used_by_unused_function(
+        "local function foo() end local function bar() foo() end"
+    ) => "",
+    remove_local_but_keep_function_call("local foo = print('hello')") => "print('hello')",
+    remove_two_locals_but_keep_function_call(
+        "local foo, bar = print('hello')"
+    ) => "print('hello')",
+    remove_two_locals_and_a_value_but_keep_function_call(
+        "local foo, bar = print('hello'), false"
+    ) => "print('hello')",
+    remove_one_local_and_keep_side_effect_value(
+        "local foo = true, print('hello')"
+    ) => "print('hello')",
+    remove_first_local_and_keep_multiple_side_effect_values(
+        "local foo, bar = true, false, print('hello'), print('bye') return bar"
+    ) => "local bar = false, print('hello'), print('bye') return bar",
+    remove_second_local_and_keep_multiple_side_effect_values(
+        "local foo, bar = true, false, print('hello'), print('bye') return foo"
+    ) => "local foo = true, print('hello'), print('bye') return foo",
+    keep_only_used_constants(
+        "local a, b = true, false return b"
+    ) => "local b = false return b",
+    remove_unused_after_last_used_in_tuple_extract(
+        "local a, b, c = ... return b"
+    ) => "local a, b = ... return b",
+    remove_variable_before_tuple_extract(
+        "local a, b, c = true, ... return b"
```

Comment:
is this correct?

Reply:
I had to think about it for a moment again but yes 😄Equivalent to:
```lua
local a = true
local b, c = ...
return b
```
So you can get rid of `a` and `c`